impl/src/main/java/org/ehcache/config/builders/CacheManagerBuilder.java

```diff
@@ -320,25 +285,20 @@ T cast(EhcacheManager ehcacheManager) {
    * @see PooledExecutionServiceConfigurationBuilder
    */
   public CacheManagerBuilder<T> withDefaultEventListenersThreadPool(String threadPoolAlias) {
-    CacheEventDispatcherFactoryConfiguration config = configBuilder.findServiceByClass(CacheEventDispatcherFactoryConfiguration.class);
-    if (config == null) {
-      return new CacheManagerBuilder<>(this, configBuilder.addService(new CacheEventDispatcherFactoryConfiguration(threadPoolAlias)));
-    } else {
-      ConfigurationBuilder builder = configBuilder.removeService(config);
-      return new CacheManagerBuilder<>(this, builder.addService(new CacheEventDispatcherFactoryConfiguration(threadPoolAlias)));
-    }
+    return using(new CacheEventDispatcherFactoryConfiguration(threadPoolAlias));
   }
 
   /**
    * Adds a {@link ServiceCreationConfiguration} to the returned builder.
    * <p>
-   * These configurations are used to load services and configure them at creation time.
+   * These configurations are used to load services and configure them at creation time.  This method will remove any
+   * existing configuration incompatible with the given configuration, before adding the new configuration.
```

Comment:
Not sure that "incompatible with" is sufficient; might need to use "duplicating or incompatible with" since the updated implementation removes configurations of the same type.

Reply:
Sure it is... by default configurations are "incompatible" with other instances of the same type.

Reply:
I have extended the "compatibleWith" concept to creation configurations.  I'll document what this means regarding typing in the javadoc.